hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/utils/HddsVolumeUtil.java

```diff
@@ -215,6 +227,17 @@ public static boolean checkVolume(HddsVolume hddsVolume, String scmId,
         logger.error("Unable to create ID directory {} for datanode.", idDir);
         return false;
       }
+
+      if (VersionedDatanodeFeatures.SchemaV3.chooseSchemaVersion()
```

Comment:
Please use a helper function to detect whether SchemaV3 is enabled. And user this helper function in all places where need a detection.  

I discussed the feature with Arpit today. His suggestion is in the first release version, we disable this feature by default.  User can turn on it manually. So other than this VersionedDataFeatures check, we may need a property, which disable, or enable this feature, just like we did internally. 
 When this feature become mature enough, we deprecate this property and make the feature always enabled.

Reply:
Yes, of course, I can't agree more with Arpit.
It would be easy for me to add this config back to the codes.
Having this config item also benefits the compatibility test as I tested this feature internally.